### PR TITLE
start integration test cluster on int-test

### PR DESCRIPTION
**Pull request by Nesurion**
Start a coreos machine to run the test against a real fleet.
The integration test machine gets started before the test and is destroyed after.

### CODE REVIEW

File: int-tests/vagrant/user-data [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#cloud-config
+
+---
+coreos:
+  etcd2:
+    discovery: https://discovery.etcd.io/e2e0adc537a5edb6e882ba0e2c65c234
```
Comment (JosephSalisbury) [inline int-tests/vagrant/user-data]: Let's not commit the discovery URL, you can leave this blank (e.g: `discovery:`)

File: int-tests/vagrant/user-data [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    command: start
+  - name: fleet.service
+    command: start
+  - name: fleet.socket
+    drop-ins:
+    - name: 30-ListenStream.conf
+      content: |
+        [Socket]
+        ListenStream=0.0.0.0:49153
+  - name: docker-tcp.socket
```
Comment (JosephSalisbury) [inline int-tests/vagrant/user-data]: Why do we need this?

File: Makefile [hunk 3]
```diff
@@ -87,6 +88,9 @@ ci-test: $(SOURCE) VERSION .gobuild
 # Set fleet endpoint to a fleet API endpoint available to the container.
 int-test: $(BIN) $(INT_TESTS)
 	@echo Running integration tests
+	@echo Starting coreos integration test machine
```
Comment (JosephSalisbury) [inline Makefile]: CoreOS

File: Makefile [hunk 4]
```diff
@@ -95,3 +99,5 @@ int-test: $(BIN) $(INT_TESTS)
 		-v $(INT_TESTS_PATH):$(INT_TESTS_PATH) \
 		zeisss/cram-docker \
 		-v $(INT_TESTS_PATH)
+	@echo destroying the integration test machine
```
Comment (JosephSalisbury) [inline Makefile]: Destroying

**Comment by Nesurion** [comment]
@giantswarm/cena RFR

File: Makefile [hunk 5]
```diff
@@ -87,7 +88,10 @@ ci-test: $(SOURCE) VERSION .gobuild
 # Set fleet endpoint to a fleet API endpoint available to the container.
 int-test: $(BIN) $(INT_TESTS)
 	@echo Running integration tests
-	docker run \
+	@echo Starting CoreOS integration test machine
+	cd $(VAGRANT_PATH) && vagrant up
+	sleep 10
+	-docker run \
```
Comment (zeisss) [inline Makefile:97]: please add a note why we prefix this with a dash

File: README.md [hunk 6]
```diff
@@ -19,6 +19,18 @@ Simply run the binary like `fleetctl`. See help usage for more information.
 inagoctl -h
 ```
 
+## Running integration tests
+
+Runnning `make int-test` will execute the integration test suit. The tests need
+to run against fleet. Starting and destroying the test machine is done via the
+make target, so you don't need to start a machine yourself.
```
Comment (zeisss) [inline README.md]: which target? `int-test`?

File: README.md [hunk 7]
```diff
@@ -19,6 +19,18 @@ Simply run the binary like `fleetctl`. See help usage for more information.
...
+
+### Integration Test Machine Configuration
+
+1. Copy `inago/int-tests/vagrant/user-data.sample` to `inago/int-tests/vagrant/user-data`
```
Comment (zeisss) [inline README.md]: Can we add a reasoning WHY I need to do this?

**Comment by zeisss** [comment]
LGTM

**Pull request merged**